testsuite/MDAnalysisTests/coordinates/test_mol2.py

```diff
@@ -184,3 +184,12 @@ def test_mol2_write_NIE():
     u = make_Universe(trajectory=True)
     with pytest.raises(NotImplementedError):
         u.atoms.write(outfile)
+
+def test_mol2_multi_write():
+    # see: gh-2678
+    with tempdir.TempDir() as tmpdir:
```

Comment:
It doesn't seem to leave any files behind and there are other uses in our code base--what's the justification for using one over the other?

I had assumed our own solution was there for a reason.

Reply:
i.e., if `testsuite/MDAnalysisTests/tempdir.py` shouldn't be used, may it should be removed or deprecated or whatever

Reply:
I see it is just copied from other examples in the same module, so if a broader cleanup is in order that's probably a separate task, though I'm happy to change this specific case...

Reply:
> I had assumed our own solution was there for a reason.

Yeah, I'm not sure why it's preferred over the fixture. It looks like a pretty straightforward context manager, although there's a comment that created files might not be deleted on Windows if they're still open. I guess one advantage it has is that you can add a prefix or suffix to the directory name, although I don't think that is ever used in the test suite. There's also a decorator that is used twice. Either way I think pytest's fixture can probably replace all instances of the `tempdir` module.

I also like ``with tmpdir.as_cwd()`` because it's clearer what is happening. You don't need to ``outfile = os.path.join(str(tmpdir), 'group1.mol2')`` because you've changed directories, but that's not immediately obvious.